specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2021-09-01-preview/appplatform.json

```diff
@@ -3695,7 +3707,62 @@
           },
           "readOnly": true
         }
-      }
+      },
+      "discriminator": "type"
+    },
+    "KeyVaultCertificateProperties": {
+      "description": "Properties of certficate imported from key vault.",
+      "required": [
+        "type",
```

Comment:
We are making "type" required and also mark it as "readonly=true", will it conflict?

Reply:
Or it will actually be auto calculated to be "KeyVaultCertificateProperties" or "ContentCertificateProperties"